mantidproject/mantid
# Refactor ALFInstrumentView

**Description of work.**
This PR refactors the code used for the ALFView interface instrument view in order to simplify the code. The `BaseCustomInstrumentView` and `ALFCustomInstrumentView` have been combined into the `ALFInstrumentView` because this greatly reduces the complexity of the file structure, and the `BaseCustomInstrumentView` is not used for anything else. 

**To test:**
Load data ALF82301
Click on Pick tab
Expand Rebin section. Filter by d-spacing range 5.5,0.01,6
Click on Select whole tube icon
Right click on the centre yellow square in the instrument view and select Extract Single Tube
Press Update Estimate button
Press Fit button
See that the peak centre value changes and a green "Fit success" label appears

*There is no associated issue.*

*This does not require release notes* because **there are no visible changes to the interface**

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
